co2eq/purchase/index.js

```diff
@@ -105,25 +108,56 @@ export function modelCanRun(activity) {
 
   return false;
 }
+
 function correctWithParticipants(footprint, participants) {
   return footprint / (participants || 1);
 }
+
 function extractEur({ costAmount, costCurrency }) {
   return (costAmount && costCurrency)
     ? convertToEuro(costAmount, costCurrency)
     : null;
 }
 
+function CPIConversion(eurAmount, referenceYear, countryCodeISO2, datetime) {
+  if ((!eurAmount) || (!datetime)) {
+    return eurAmount;
+  }
+
+  const currentDateIndicator = datetime.getFullYear();
+
+  let CPIcurrent;
+  if (countryCodeISO2 && consumerPriceIndex[COUNTRY_CPI_INDICATOR][countryCodeISO2][currentDateIndicator]) {
+    CPIcurrent = consumerPriceIndex[COUNTRY_CPI_INDICATOR][countryCodeISO2][currentDateIndicator];
+  } else {
+    CPIcurrent = consumerPriceIndex[AVERAGE_CPI_COUNTRY_INDICATOR][currentDateIndicator];
+  }
+
+  let CPIreference;
+  if (countryCodeISO2 && consumerPriceIndex[COUNTRY_CPI_INDICATOR][countryCodeISO2][referenceYear]) {
+    CPIreference = consumerPriceIndex[COUNTRY_CPI_INDICATOR][countryCodeISO2][referenceYear];
+  } else {
+    CPIreference = consumerPriceIndex[AVERAGE_CPI_COUNTRY_INDICATOR][referenceYear];
+  }
```

Comment:
What if we don't have a CPI for the `referenceYear `? Raising an error would be my recommendation.

Reply:
Indeed if there is no CPI for average (which could happen in 2021 if we don't update anything until then or if someones declares an activity before 2011) we should raise an error, but shouldn't we use the average as fallback if there is no cpi for the country for that given year?

Reply:
Good catch - yes let's do that